src/I18n/Date.php

```diff
@@ -130,165 +130,6 @@ class Date extends MutableDate implements JsonSerializable
-                $months = 11;
-                $years--;
-            }
-
-            if ($months >= 12) {
-                $years = $years + 1;
-                $months = $months - 12;
-            }
-
-            if ($days >= 7) {
-                $weeks = floor($days / 7);
-                $days = $days - ($weeks * 7);
-            }
-        } else {
-            $years = $months = $weeks = 0;
-            $days = floor($diff / 86400);
-
-            $diff = $diff - ($days * 86400);
-
-            $hours = floor($diff / 3600);
-            $diff = $diff - ($hours * 3600);
-
-            $minutes = floor($diff / 60);
-            $diff = $diff - ($minutes * 60);
-            $seconds = $diff;
-        }
-
-        $fWord = $options['accuracy']['day'];
-        if ($years > 0) {
-            $fWord = $options['accuracy']['year'];
-        } elseif (abs($months) > 0) {
-            $fWord = $options['accuracy']['month'];
-        } elseif (abs($weeks) > 0) {
-            $fWord = $options['accuracy']['week'];
-        } elseif (abs($days) > 0) {
-            $fWord = $options['accuracy']['day'];
-        }
-
-        $fNum = str_replace(['year', 'month', 'week', 'day'], [1, 2, 3, 4], $fWord);
-
-        $relativeDate = '';
-        if ($fNum >= 1 && $years > 0) {
-            $relativeDate .= ($relativeDate ? ', ' : '') . __dn('cake', '{0} year', '{0} years', $years, $years);
-        }
-        if ($fNum >= 2 && $months > 0) {
-            $relativeDate .= ($relativeDate ? ', ' : '') . __dn('cake', '{0} month', '{0} months', $months, $months);
-        }
-        if ($fNum >= 3 && $weeks > 0) {
-            $relativeDate .= ($relativeDate ? ', ' : '') . __dn('cake', '{0} week', '{0} weeks', $weeks, $weeks);
-        }
-        if ($fNum >= 4 && $days > 0) {
-            $relativeDate .= ($relativeDate ? ', ' : '') . __dn('cake', '{0} day', '{0} days', $days, $days);
-        }
-
-        // When time has passed
-        if (!$backwards && $relativeDate) {
-            return sprintf($options['relativeString'], $relativeDate);
-        }
-        if (!$backwards) {
-            $aboutAgo = [
-                'day' => __d('cake', 'about a day ago'),
-                'week' => __d('cake', 'about a week ago'),
-                'month' => __d('cake', 'about a month ago'),
-                'year' => __d('cake', 'about a year ago')
-            ];
-
-            return $aboutAgo[$fWord];
-        }
-
-        // When time is to come
-        if (!$relativeDate) {
-            $aboutIn = [
-                'day' => __d('cake', 'in about a day'),
-                'week' => __d('cake', 'in about a week'),
-                'month' => __d('cake', 'in about a month'),
-                'year' => __d('cake', 'in about a year')
-            ];
-
-            return $aboutIn[$fWord];
-        }
-
-        return $relativeDate;
+        return (new RelativeTimeFormatter($this))->dateAgoInWords($options);
```

Comment:
I think at the end it will have to be static. The idea that i had was to use the `ChronosInterface::diffFormatter()` object for doing this. But now that I give a second look to that method, I realize that it will be quite difficult to globally use the same formatter for all objects (with the same translator, for example)